src/ChplPaths.ts

```diff
@@ -195,3 +195,50 @@ export function getWorkspaceFolder(): vscode.WorkspaceFolder | undefined {
   const doc = editor.document;
   return vscode.workspace.getWorkspaceFolder(doc.uri);
 }
+
+export function getEnvAffectingChapel(): Map<string, string> {
+  const globs = [
+    "CHPL_.*",
+    "CHPL_RT_.*",
+    "FI_.*",
+    "GASNET_.*",
+    "SLURM_.*",
+    "PBS_.*",
+    "QTHREAD_.*",
+    "QT_.*",
+    "ASAN_OPTIONS",
+  ];
+  // TODO: consider including?
+  // PKG_CONFIG_PATH
+  // HUGETLB_NO_RESERVE
+  // MASON_REGISTRY
+  // SPACK_ROOT
+  // PYTHON*
+  // LD_LIBRARY_PATH
+  // DYLD_LIBRARY_PATH
+  const regex = new RegExp(`^(${globs.join('|')})$`)
+
+  let env = new Map();
+  for (const e in process.env) {
+    if (regex.test(e)) {
+      env.set(e, process.env[e] ?? "");
+    }
+  }
+
+  // special handling for CHPL_HOME and CHPL_DEVELOPER
+  const chplhome = cfg.getChplHome();
+  if (chplhome != undefined || chplhome !== "") {
```

Comment:
This should be and nor or.

```suggestion
  if (chplhome != undefined && chplhome !== "") {
```

Reply:
😄 I actually hit this subtle bug while working on the next steps from this PR, and had forgotten to update this PR. Good catch!